nodebuilder/header/header.go

```diff
@@ -4,32 +4,78 @@ import (
 	"context"
 
 	"github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
 )
 
 // Module exposes the functionality needed for querying headers from the network.
 // Any method signature changed here needs to also be changed in the API struct.
 //
 //go:generate mockgen -destination=mocks/api.go -package=mocks . Module
 type Module interface {
+	// Head returns the ExtendedHeader of the chain head.
+	Head(context.Context) (*header.ExtendedHeader, error)
+
+	// Get returns the header of the given hash from the node's header store.
+	Get(ctx context.Context, hash libhead.Hash) (*header.ExtendedHeader, error)
+	// GetVerifiedRangeByHeight returns the given range [from:to) of ExtendedHeaders
+	// from the node's header store and verifies that the returned headers are
+	// adjacent to each other.
+	GetVerifiedRangeByHeight(
+		ctx context.Context,
+		from *header.ExtendedHeader,
+		to uint64,
+	) ([]*header.ExtendedHeader, error)
 	// GetByHeight returns the ExtendedHeader at the given height, blocking
 	// until header has been processed by the store or context deadline is exceeded.
 	GetByHeight(context.Context, uint64) (*header.ExtendedHeader, error)
-	// Head returns the ExtendedHeader of the chain head.
-	Head(context.Context) (*header.ExtendedHeader, error)
+
 	// IsSyncing returns the status of sync
 	IsSyncing(context.Context) bool
```

Comment:
We should return the state of syncer instead, by which we can determine if its syncing or not

Reply:
In follow-up PR - I think we should have both. `IsSyncing` is just a bool and doesn't require user to import sync.State

Reply:
There is no reason to have one more method if there is another one from which we can simply derive the same information

Reply:
I don't get why would we do this in another PR if the goal of this PR is to extend the header module

Reply:
> There is no reason to have one more method if there is another one from which we can simply derive the same information

for convenience - the same way we will have blob module. I think people would care more about whether node is Syncing than to have the entire sync state.

> I don't get why would we do this in another PR if the goal of this PR is to extend the header module

Because we already have this method `IsSyncing` and i want to keep it as is, and then have another method to return all of syncstate which is an additional endpoint, so i'd rather do it later.

Reply:
In private, we agreed to remove it